src/classes/RD2_StatusMapper_TEST.cls

```diff
@@ -36,7 +36,8 @@ public with sharing class RD2_StatusMapper_TEST {
     private static final Set<String> READ_ONLY_STATUS_VALUES = new Set<String>{
```

Comment:
Instead of updating the Set every time a new picklist is created, probably it can be set as `RD2_StatusMapper. READ_ONLY_STATUS_TO_STATE.keySet()`?